RRS-93 Create a review page

Ailam7 opened the pull request:
Added all the fields required for the form to create a new review, and applied styling according to the prototypes.

Ailam7 committed:
Added Appropriate Fields to Create Review page according to prototypes

Ailam7 committed:
Added Appropriate Fields to Create Review page according to prototypes

Ailam7 committed:
Added additional styling to resemble prototype more closely

Ailam7 committed:
Added additional styling to resemble prototype more closely

LukeHosk commented on README.md:
Is this intentional?

LukeHosk committed:
update create review page to use search parameters for the propertyId and address

LukeHosk commented on app/reviews/create/page.tsx:
```diff
@@ -1,77 +1,172 @@
-          className="border p-2 rounded-md"
-          name="review_body"
-          id="review_body"
+        <label htmlFor="proof_of_tenancy">Upload proof of tenancy</label>
```
I thought we said we were only going to require proof of tenancy if a dispute is raised?

We definitely said we weren't handling files until S3. Can we leave this out for now?

Ailam7 replied:
Yes you're right I forgot about this. I'll fix this in the next commit. 

Should I remove the field for uploading images as well for now?

LukeHosk replied:
You can probably just comment that one  out for now since we do want to add it later, but yes, I’d remove it from the initial prototype.

you could also put a placeholder there that explicitly acknowledges ‘image uploads coming soon’ or something to that effect if you want to preserve the layout

LukeHosk commented on app/reviews/create/page.tsx:
```diff
@@ -1,77 +1,172 @@
-        ></textarea>
+        />
+
+        <h2 className="text-xl font-semibold">Landlord Ratings</h2>
```
I also thought we said we were going to have a single 'Landlord Rating' rather than splitting it like this.

The backend is currently only set up to accept a single rating. Have you tested submitting this form? Pretty sure it will just error as no 'landlord_rating' is provided.

Ailam7 replied:
My bad, will fix this 👍

LukeHosk commented on app/reviews/create/page.tsx:
```diff
@@ -1,77 +1,172 @@
 
         {state.message && <p>{state.message}</p>}
-        {state.errors && <p>{JSON.stringify(state.errors, null, '\t')}</p>}
+        {state.errors && <p>{JSON.stringify(state.errors, null, "\t")}</p>}
```
I think we want to show errors more gracefully to users than the raw JSON

LukeHosk commented on app/reviews/create/page.tsx:
```diff
@@ -1,77 +1,172 @@
-        className="flex flex-col gap-2 pleace-items-center max-w-prose w-full border p-4 rounded-md"
+        className="flex flex-col gap-2 pleace-items-center w-full border p-4 rounded-md mb-8"
         action={dispatch}
+        style={{ backgroundColor: "#778378", color: "#222" }}
```
are these colours supposed to cover the whole page as in the designs?

we probably want to [add them as custom colours in the tailwind theme](https://tailwindcss.com/docs/customizing-colors#adding-additional-colors)

and then set the background in the root layout so that it covers the whole page, as I assume you do not want it to look like this: 

<img width="1465" alt="Screenshot 2024-02-17 at 22 01 00" src="https://github.com/CM20254-Group-7/Rental_Review/assets/74842299/186d70d1-3bb5-4b81-b23c-8c95e9f0a057">

Ailam7 replied:
So when I tried applying the color to the whole div, it applied the color only to the small column occupied by the div rather than the whole page. So as you say we should change it in the root layout if we agree all pages should have this background color. 

It looked very weird when just the div had the background color, so I kept it only on the form for now, but later when we change the color of the whole page we should change the bg-color to the whole div rather than just the form. I'll change the style from just the form to the whole div in the next commit.

LukeHosk replied:
I'm pretty sure we have it as the background colour in all the prototype mockups, so it would make sense to set in the root layout.

If these are standard colours, it would also make sense to use css variables for them. See [here](https://github.com/CM20254-Group-7/Rental_Review/blob/c0716203d141e149900cede1780e7503683e9b5e/app/globals.css#L5-L21) for setting variables for the light and dark theme and [here](https://github.com/CM20254-Group-7/Rental_Review/blob/c0716203d141e149900cede1780e7503683e9b5e/tailwind.config.js#L9-L16) for extending tailwind with those variables.

You can either replace the existing background variable values, or create a one e.g. 'primary'
 
This is potentially worth adding a new task as part of a 'consistent UI design' story for especially since it will involve going over each existing page and ensuring compatability

LukeHosk replied:
You might want to have a look at something like realtime colours. Which can export themes in this format:

<img width="1261" alt="Screenshot 2024-02-20 at 14 39 57" src="https://github.com/CM20254-Group-7/Rental_Review/assets/74842299/275f04fe-795d-4c1e-93b0-8eaad701f9ef">

Have a look at this [example I made with mostly colours from the prototypes](https://www.realtimecolors.com/?colors=212121-778378-dbb866-ededed-443dff&fonts=Poppins-Poppins)

LukeHosk replied:
dealing with this in separate tasks

LukeHosk committed:
bugfix: extract client-side code to seperate function

Ailam7 committed:
Removed field for uploading proof of tenancy

Ailam7 committed:
Commented out image uploading for the time being

Ailam7 committed:
Following changes to input fields:

Removed specific landlord ratings and changed to one overall landlord rating
Added option for a property rating
Added field for a review title, which was in our development requirements

Ailam7 committed:
Re-added review date field as backend was setup for it.
Modified actions.ts to include review title in the backend

Ailam7 committed:
Merge branch 'Create_A_Review_Page' of https://github.com/CM20254-Group-7/Rental_Review into Create_A_Review_Page

I didn't realize a new ReviewCreationForm.tsx was created and its caused a mess when syncing. I've undid the previous commits and I'll redo the changes from this commit onwards.

Ailam7 committed:
Made the appropriate changes:
Removed specific landlord fields
Added option for a property rating
Added option for a review title
Modifed actions.ts to include review title

Ailam7 committed:
Added review title column to supabase

Ailam7 committed:
Modified state.errors to handle errors more gracefully

OttoChu committed:
Squashed commit of the following:

commit e27a07d939b5deb0d9b04bb726d16b7774374cae
Author: Otto Chu <96123119+OttoChu@users.noreply.github.com>
Date:   Fri Feb 23 10:17:41 2024 +0000

    RRS-108 Create Landlord Profile Page (#27)

    * Created example landlord profile page

    * Added check user logged in

    * Able to get real data from database

    * Fixed few bugs

    - Still show profile if the landlord owns no properties
    - used noFound when landlord doesn't exist
    - Added bio and made what to change more understandable

    * Added the dynamic routing feature

    * Changed where landlord details are stored

    * fields already removed from user table

    * type file index issue fix

    ---------

    Co-authored-by: Luke Hoskins <luke@hoskins.cloud>

commit 64f3ab723da0d77260e05a1d394f53bc60f071b0
Author: Luke Hoskins <74842299+LukeHosk@users.noreply.github.com>
Date:   Thu Feb 22 15:45:25 2024 +0000

    RRS-150 New User Profiles V2 (#33)

    * chore: update environment variable files to include .env and remove .env.local, update values in .env file

    * chore: exclude auth sessions from playwright in git-ignore

    * install dotenv

    * use env and setup files in playwright config

    * setup auth states for each test user wheh initialising tests

    * created example tests with various auth states

    * chore: add sample users to the database seed

    * create new user profile migration (with the trigger this time)

    * extract helper functions to own file

    * setup auth state & user profile unit testing

    * fix missing lines from copy paste

    * type update

commit e66f40a842930a1c8aa3ba7df58891dc26bf0a52
Author: Luke Hoskins <74842299+LukeHosk@users.noreply.github.com>
Date:   Thu Feb 22 13:34:58 2024 +0000

    chore: update workflow names and descriptions for clarity and consistency (#34)

commit 7768fc7d4556f9175e20f80bdb976a50caf16d7e
Author: Luke Hoskins <74842299+LukeHosk@users.noreply.github.com>
Date:   Thu Feb 22 12:20:31 2024 +0000

    Revert "create db function to create a basic user profile and trigger it when…" (#32)

    This reverts commit c354fd617cfec163f2a5ac47bc8d2e0e61bfe081.

commit ea9c49193585db990536495ef8b051d66f7a70d9
Author: Luke Hoskins <74842299+LukeHosk@users.noreply.github.com>
Date:   Wed Feb 21 21:53:48 2024 +0000

    handle not found on property page (#31)

commit c354fd617cfec163f2a5ac47bc8d2e0e61bfe081
Author: Luke Hoskins <74842299+LukeHosk@users.noreply.github.com>
Date:   Wed Feb 21 16:43:50 2024 +0000

    create db function to create a basic user profile and trigger it when a user signs up (#14)

commit b29b8e5bd28da4fc3c3606d52c5c8eddb5050989
Author: Otto Chu <96123119+OttoChu@users.noreply.github.com>
Date:   Tue Feb 20 18:04:06 2024 +0000

    RRS-148 create server action to add property ownership (#22)

    * Added basic claim property actions

    * tutorial notes

    * Started to add checks

    * Completed the checks to claim property

    I have just throw errors when the property has already been claimed. Might need to change it later once the landlord table stuff is done

    * create basic property claiming form for testing

    * update function definition to:
    - use bound values for the IDs
    - correctly accept and return the state

    * use validation on action inputs & use Date instead of string for comparisons

    * extend validation to ensure dates are in the past

    * Issue identified: function does not return a state on all paths.

    start to fix by adding a success message after adding data

    * extract data setting to seperate function

    * removed if-not-null check on data that is already guarunteed not to be null.

    set property ownership immediatly if no existing ownership records are found

    * planning: define plan for claim checking

    * move ownership record creation from within loop to after (function would have tried to make a new record for every existing claim it didn't overlap).

    Create function to close an open claim before creating the new one

    allow action to track if there is an open claim and call the function created above to close it if there is

    * set claim creation to use `insert` not `update`

    * landlord_id should not be nullable

    * style: remove un-necessary else-ifs after conditions that cause the function to return. flattens code and increases readability.

    use standard formatting for all return statements

    * fixed it

    * actually fixed it

    * Update supabase.types.ts

    * fix index issue

    ---------

    Co-authored-by: Luke Hoskins <luke@hoskins.cloud>

commit 2dc379b3cac3b4d174eb15e3336b9e34f9d0c39f
Author: Matthew <43332061+mattg42@users.noreply.github.com>
Date:   Tue Feb 20 17:28:38 2024 +0000

    RRS-45 Create property detail pages (#26)

    * Moved StarRatings to its own file

    * Decreased size of stars

    * Basic property page

    * Changed to use dynamic routing

commit 5d4b8771dd5d7f75ceb69a6c48b4081e4253c87d
Author: Luke Hoskins <74842299+LukeHosk@users.noreply.github.com>
Date:   Tue Feb 20 17:18:03 2024 +0000

    RRS-154 Enable frontend & E2E unit testing (#19)

    * install playwright

    * rename frontend testing github action

    * start supabase before running frontend tests

    * change supabase start command

    * run on all PRs

    * configure playwright to start required servers

    * create test tests that use our app

    * fix: setup environonment variables correctily

    * rename example test

    * remove un-necessary duplication of tasks

commit 8d5a5538891e58af705471148a043d39223d1272
Author: Mad-Mad-Cat <84139460+Mad-Mad-Cat@users.noreply.github.com>
Date:   Tue Feb 20 17:15:11 2024 +0000

    Rrs 29 create property search page (#25)

    * PropertiesDB readable, displayed, home button

    * console log removed + renaming

    * uncommented link in login page

    ---------

    Co-authored-by: Jean-Luc <johnnylucvan@gmail.com>

commit 4550387513e64f292ae29081b29032202bcd278f
Author: euand1818 <160054443+euand1818@users.noreply.github.com>
Date:   Tue Feb 20 17:11:43 2024 +0000

    re generated types (#24)

commit 7976dad941c020653282eab9b0b9f6f06558f651
Author: Luke Hoskins <74842299+LukeHosk@users.noreply.github.com>
Date:   Tue Feb 20 17:00:17 2024 +0000

    RRS-153 Enable Supabase unit testing (#18)

    * chore: add GitHub Actions workflow for running database tests on pull request

    * test: add example unit test which checks if the auth.users table has an id field

commit c0716203d141e149900cede1780e7503683e9b5e
Author: Luke Hoskins <74842299+LukeHosk@users.noreply.github.com>
Date:   Tue Feb 20 14:26:23 2024 +0000

    Review Detail Component (#15)

    * RRS-69 Create component

    * RRS-72 Property Rating

    * RRS-71 Landlord Rating

    * RRS-70 Review Ratings

    * RRS-68 Review Text

    * added missing icon imports

commit 1c481af86dc9d3cc1fd67ae51377ab9319b1d06f
Merge: fa131a3 4b2df99
Author: euand1818 <160054443+euand1818@users.noreply.github.com>
Date:   Mon Feb 19 14:42:39 2024 +0000

    Merge pull request #17 from CM20254-Group-7/RRS-128-store-landlord-details-in-DB

    RRS-128 store landlord details in db

commit 4b2df9941152e151ac94eb1160080b32b739d78c
Author: Euan <euand1818@gmail.com>
Date:   Mon Feb 19 14:40:50 2024 +0000

    re-re created types file

commit 69593c31ca5ff31345f0b9e336f8f4162331b7e0
Merge: 9e14b59 807602e
Author: Euan <euand1818@gmail.com>
Date:   Mon Feb 19 14:40:40 2024 +0000

    Merge branch 'RRS-128-store-landlord-details-in-DB' of https://github.com/CM20254-Group-7/Rental_Review into RRS-128-store-landlord-details-in-DB

commit 9e14b591172a428f6cf4254609e2c5510793471d
Author: Euan <euand1818@gmail.com>
Date:   Mon Feb 19 14:33:00 2024 +0000

    Generated types

commit 7bd6de2d166f0ab2276f3b70dfdbfa32672d6ae0
Author: Euan <euand1818@gmail.com>
Date:   Mon Feb 19 14:30:36 2024 +0000

    bugs

commit c972497ae5d42819162286fc1d903ce33485da6e
Author: Euan <euand1818@gmail.com>
Date:   Mon Feb 19 14:29:34 2024 +0000

    changed naming of landlord id to user_id

commit fa131a3646b2c97279cf5247512a9c74665451bb
Author: mrepicness888 <150257321+mrepicness888@users.noreply.github.com>
Date:   Mon Feb 19 14:10:36 2024 +0000

    RRS-110 Create login page

    * new test commit

    * Revert "new test commit"

    This reverts commit 413c8dd524e0acc770868a9647a1b63d4e66cebd.

    * Revert "New test"

    This reverts commit 976512dc84f15086532473a6739f822e9b03b842.

    * refactor: move signIn and signUp functions to separate file and remove unused imports

    * use zod for input validation and states instead of search params for messaging

    * refactor: extract login form into a seperate component

    * chore: refactor sign-in form to use a reusable Form component

    * split out sign up form

    * update styping to work better with two forms

    * show detailed errors inline

    * refactor login and signup forms to use reusable Field components for email and password inputs

    * give more specific error when email is taken

    ---------

    Co-authored-by: Luke Hoskins <luke@hoskins.cloud>

commit 807602e69d1205113be48d60f0d98e62597b78b3
Author: Luke Hoskins <luke@hoskins.cloud>
Date:   Sun Feb 18 19:40:27 2024 +0000

    regenerate database types after landlord details changed

commit af1ff45d5bf6a6d240e627cdbefeeffbe4d6f14a
Author: Euan <euand1818@gmail.com>
Date:   Sun Feb 18 18:11:45 2024 +0000

    changed back to id

commit 9f11fc5a4d166263a2a831538f28fe22c539d038
Author: Euan <euand1818@gmail.com>
Date:   Sun Feb 18 17:57:39 2024 +0000

    add fields to landlord_private_profiles and landlord_public_profiles

Ailam7 committed:
Tried wierd bug fix that didn't work

OttoChu committed:
Fixed the problem with creating reviews

OttoChu committed:
Update .gitignore

OttoChu committed:
Forgot to gen types in the previous commit

LukeHosk committed:
re-apply: Added text to specify tenancy period would be in months, as we are taking it as an int

LukeHosk committed:
Merge remote-tracking branch 'origin/development' into Create_A_Review_Page
